lib/Doctrine/DBAL/Migrations/Configuration/Configuration.php

```diff
@@ -816,6 +822,24 @@ public function generateVersionNumber(\DateTimeInterface $now=null)
     }
 
     /**
+     * Explicitely opens the database connection. This is done to play nice
+     * with DBAL's MasterSlaveConnection. Which, in some cases, connects to a
+     * follower when fetching the executed migrations. If a follower is lagging
+     * significantly behind that means the migrations system may see unexecuted
+     * migrations that were actually executed earlier.
+     *
+     * @return bool The same value returned from the `connect` method
+     */
+    protected function connect()
+    {
+        if ($this->connection instanceof MasterSlaveConnection) {
+            return $this->connection->connect('master');
```

Comment:
Good question. `MasterSlaveConection` [only accepts `master` and `slave` as arguments to connect](https://github.com/doctrine/dbal/blob/69b57859c8674f2c8ba3798a2541ac18128a0bdb/lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php#L147-L149), so we should be okay.